### PR TITLE
[CUJ2] add nightly tests for running 500GB ray train 

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

We run 500GB ray preprocessing and train in e2e tests.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/dataset/ray_sgd_training_app.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+python:
+  pip_packages: []
+  conda_packages: []
+
+post_build_cmds:
+    - pip install llvmlite --ignore-installed
+    - pip install "ray[tune,data,default] @ https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-2.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl" jupyter torch awscli pyarrow tensorboard boto3 fsspec pickle5 dask dask_ml s3fs mlflow torchvision
+    - pip uninstall aiohttp -y
+    - pip install aiohttp==3.7
+    - mkdir -p data inference models runs
```
Comment (rkooo567) [inline release/nightly_tests/dataset/ray_sgd_training_app.yaml:12]: New line!

**Review by rkooo567 — COMMENTED** [review]
Didn't look at the test script. Should I look at it, or you will ask other people to do? 

Also, we need to update `build_pipeline.py`. And is it possible we build a smoke test for this with smaller compute requirement? I think we should avoid this much heavy test twice a day. (I even feel like we should run this once a week give how expensive the test will be...)

Example; 

https://github.com/ray-project/ray/blob/dc51af798c3cdb79aceaf9f0c1ca63fc1d2b12b5/release/.buildkite/build_pipeline.py#L78
https://github.com/ray-project/ray/blob/dc51af798c3cdb79aceaf9f0c1ca63fc1d2b12b5/release/nightly_tests/nightly_tests.yaml#L166

**Comment by matthewdeng** [comment]
For latest cluster env [change](https://github.com/ray-project/ray/pull/20195/commits/d7dcbe6e88d58e05c60495b6f38ea330c26ee109) was run [here](https://console.anyscale.com/o/anyscale-internal/projects/prj_uYugyQ1qnvQAT7PQF2DYktFk/sessions/ses_WjHFEnrCagZ13JNCpi13Nh2U?command-history-section=command_history).

**Review by matthewdeng — COMMENTED** [review]
Do you want to add the smoke test to `build_pipeline.py` to start running nightly tests?

File: release/nightly_tests/dataset/dataset_test.yaml [hunk 2]
```diff
@@ -52,3 +52,27 @@
     timeout: 4800
     prepare: python wait_cluster.py 21 2400
     script: python pipelined_training.py --epochs 2 --num-windows 15  --num-files 915 --debug
+
+- name: ray_sgd_500gb
```
Comment (matthewdeng) [inline release/nightly_tests/dataset/dataset_test.yaml:56]: This naming makes sense for the full test but not the smoke test.

File: release/nightly_tests/dataset/dataset_test.yaml [hunk 3]
```diff
@@ -52,3 +52,27 @@
...
     script: python pipelined_training.py --epochs 2 --num-windows 15  --num-files 915 --debug
+
+- name: ray_ml_ingestion
+  owner:
+    mail: "core@anyscale.com"
+    slack: "@Chen Shen"
+
+  cluster:
+    app_config: ray_sgd_training_app.yaml
+    compute_template: ray_sgd_training_compute.yaml
```
Comment (matthewdeng) [inline release/nightly_tests/dataset/dataset_test.yaml:63]: nit: Rename these all to be more consistent? e.g. `datasets_ingest_train_infer`

**Review by matthewdeng — APPROVED** [review]
🚢

**Comment by scv119** [comment]
windows is failing.

**Pull request merged**